analysis/import/SCMLogParserV2/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparserv2/parser/git/GitLogNumstatRawParserStrategy.kt

```diff
@@ -0,0 +1,89 @@
+import java.time.OffsetDateTime
+import java.util.function.Predicate
+import java.util.stream.Collector
+import java.util.stream.Stream
+import kotlin.streams.toList
+
+class GitLogNumstatRawParserStrategy : LogParserStrategy {
+    override fun creationCommand(): String {
+        return "git log --numstat --raw --topo-order --reverse -m"
+    }
+
+    override fun createLogLineCollector(): Collector<String, *, Stream<List<String>>> {
+        return LogLineCollector.create(GIT_COMMIT_SEPARATOR_TEST)
+    }
+
+    override fun parseAuthor(commitLines: List<String>): String {
+        return commitLines
+            .parallelStream() // take a look here
+            .filter { commitLine -> commitLine.startsWith(AUTHOR_ROW_INDICATOR) }
+            .map { AuthorParser.parseAuthor(it) }
+            .toList()
+            .first()
+    }
+
+    override fun parseModifications(commitLines: List<String>): List<Modification> {
+        return commitLines
+            .filter { isFileLine(it) }
+            .map { parseModification(it) }
+            .groupingBy { it.currentFilename }
+            .aggregate { _, aggregatedModification: Modification?, currentModification, _ ->
+                when (aggregatedModification) {
+                    null -> mergeModifications(currentModification)
+                    else -> mergeModifications(aggregatedModification, currentModification)
+                }
+            }
+            .values
+            .toList()
+    }
+
+    override fun parseDate(commitLines: List<String>): OffsetDateTime {
+        return commitLines
+            .filter { commitLine -> commitLine.startsWith(DATE_ROW_INDICATOR) }
+            .map { CommitDateParser.parseCommitDate(it) }
+            .first()
+    }
+
+    override fun parseIsMergeCommit(commitLines: List<String>): Boolean {
+        return commitLines.any { commitLine -> commitLine.startsWith(MERGE_COMMIT_INDICATOR) }
+    }
+
+    companion object {
+
+        private val GIT_COMMIT_SEPARATOR_TEST = Predicate<String> { logLine -> logLine.startsWith("commit") }
+        private fun isFileLine(commitLine: String): Boolean {
+            return GitLogRawParsingHelper.isFileLine(commitLine) || GitLogNumstatParsingHelper.isFileLine(commitLine)
+        }
+
+        internal fun parseModification(fileLine: String): Modification {
+            return if (fileLine.startsWith(":")) {
+                GitLogRawParsingHelper.parseModification(fileLine)
+            } else GitLogNumstatParsingHelper.parseModification(fileLine)
+        }
+
+        private fun mergeModifications(vararg a: Modification): Modification {
+            val filename = a[0].currentFilename
+            val additions = a.map { it.additions }.sum()
+            val deletions = a.map { it.deletions }.sum()
+            val type =
+                a.map { it.type }.firstOrNull { t -> t != Modification.Type.UNKNOWN } ?: Modification.Type.UNKNOWN
```

Comment:
```suggestion
                a.firstOrNull { t -> t.type != Modification.Type.UNKNOWN } ?: Modification.Type.UNKNOWN
```

Reply:
firstOrNull cant transform the list, but I take a few more min to maybe find a solution in one attempt

Reply:
```suggestion
       val tmpModification = a.firstOrNull {  modification -> modification.type != Modification.Type.UNKNOWN }
            var type = Modification.Type.UNKNOWN

            if(tmpModification != null){
                type = tmpModification.type
            }

```